Mathlib/Init/Function.lean

```diff
@@ -25,6 +26,12 @@ variable {α : Sort u₁} {β : Sort u₂} {φ : Sort u₃} {δ : Sort u₄} {ζ
 
 lemma comp_def {α β δ : Sort _} (f : β → δ) (g : α → β) : f ∘ g = fun x ↦ f (g x) := rfl
 
+attribute [eqns comp_def] comp
+
+lemma flip_def {f : α → β → φ} : flip f = fun b a => f a b := rfl
+
+attribute [eqns flip_def] flip
+
```

Comment:
I am extremely hesitant to make this change in Mathlib; if we want to change a basic behaviour like this shouldn't we coordinate with Std and Lean? I worry that it introduces a confusing variation in behaviour across the ecosystem.

Reply:
For other readers, discussion [continues on Zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/v4.2E3.2E0-rc2.20-.20comp_def.20and.20.238023/near/401647442)

Reply:
I've been convinved.

Reply:
Undoability was one of the reasons I decided to go ahead and merge. I was seeing it as a workaround to help get the version bump done sooner rather than a permanent design.